webindex/modules/data/src/test/java/webindex/data/fluo/PageLoaderTest.java

```diff
@@ -0,0 +1,63 @@
+package webindex.data.fluo;
+
+import java.net.MalformedURLException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import webindex.core.models.Page;
+import webindex.core.models.URL;
+
+@SuppressWarnings("unused")
+public class PageLoaderTest {
+
+	@Test
+	public void testUpdatePageWithNullPage() {
+		Page p = null;
+		PageLoader loader = PageLoader.updatePage(p);
+		loader = null;
+	}
+	
+	@Test(expected = IllegalArgumentException.class)
+	public void testEmptyPageThrowsIllegalArgument() {
+		Page p = Page.EMPTY;
+		PageLoader loader = PageLoader.updatePage(p);
+		loader = null;
+	}
+	
+	@Test(expected = NullPointerException.class)
+	public void testDeleteFailsWithNullUrl() {
+		try {
+			PageLoader loader = PageLoader.deletePage(null);
+		} catch (MalformedURLException e) {
```

Comment:
Can this catch be dropped?

Reply:
I don't think so because of this method signature:

https://github.com/apache/fluo-examples/blob/47f1bb42969345b444f5c0e4c8c4a36ec93aab14/webindex/modules/data/src/main/java/webindex/data/fluo/PageLoader.java#L52

Reply:
If the exception is not expected you can make the test throw exception.  If it is actually thrown then the test will fail.

Reply:
Not sure if this is correct but did a simple refactor where it catches Exception e and test if its an instance of a NPE and if yes throws NPE.

Reply:
I was thinking of the following.

```java
@Test(expected = NullPointerException.class)
public void testDeletePageThrowsNPEifNullURL() throws MalformedURLException {	
  PageLoader.deletePage((URL)null);
}
```

Reply:
Ok, I gotcha now. I fixed it to throws instead of catch, and did the same for the other methods. I added throws to the last test for clarity even though it catches all exceptions, as a hint to the reader.